### ISSUE TITLE
no-unused-vars: varsIgnorePattern not respected in array destructure

### ISSUE THREAD

**Issue by LINKIWI** [eslint/eslint#10094]
### Summary

The `no-unused-vars` rule fails to respect the `varsIgnorePattern` configuration regex for parameters destructured out of an array.

This is reproducible with the following configuration:

Parsed with `babel-eslint`

Node: 8.9.0
eslint: 4.15.0
babel-eslint: 8.2.2

### MCVE

Configuration:

```javascript
// Whitelist `_` as an identifier to not error on if unused
'no-unused-vars': ['error', {
  vars: 'all',
  args: 'after-used',
  ignoreRestSiblings: true,
  varsIgnorePattern: '^_$',
}],
```

Source:

```javascript
[[1, false]].filter(([_, value]) => value);
```

Expected behavior: 0 errors and 0 warnings.

Actual behavior:

```
$ eslint mcve.js

[...clip...]/mcve.js
  1:23  error  '_' is defined but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

**Comment by LINKIWI** [comment]
To add on this: the bug is reproducible for function parameters and object destructure identifiers as well.

```javascript
[{ a: true, b: false }].filter(({ a, _ }) => a);
```

```javascript
[{ a: true, b: false }].filter((a, _) => a);
```

In all cases:

```
$ eslint mcve.js

[...clip...]/mcve.js
  1:23  error  '_' is defined but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

**Comment by not-an-aardvark** [comment]
Thanks for the report. However, I think this is working as intended -- you should use the [`argsIgnorePattern`](https://eslint.org/docs/rules/no-unused-vars#argsignorepattern) option if you want to ignore those parameters.

**Comment by LINKIWI** [comment]
Thanks for the response - I completely missed that option in the documentation. `argsIgnorePattern` works as expected.
